src/app/auth/api.http.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { ConnectionBackend, Http, Headers, Request, RequestOptions, RequestOptionsArgs, Response } from '@angular/http';
+import { Observable } from "rxjs";
+
+import { apiToken } from "../constants";
+
+import * as _ from 'lodash';
+
+/**
+ * Wrapper for Http that appends authorization headers for requests to the API backend,
+ * and redirects unauthorized responses to login page.
+ */
+@Injectable()
+export class ApiHttp extends Http {
```

Comment:
This class feels more generic than being an Http class for an API. More like `TokenAuthHttp` maybe? Since all it really does is append a token authorization header.